addon/components/frost-toggle.js

```diff
@@ -78,42 +113,40 @@ export default Component.extend(PropTypeMixin, FrostEventsProxy, {
     return e
   },
 
+  // FIXME: jsdoc
   _preferBoolean (value) {
     if (value === 'true') return true
     if (value === 'false') return false
 
     return value
   },
 
+  // FIXME: jsdoc
   _setupAssertion () {
     assert(`Same value has been assigned to both ${this.toString()}.trueValue and ${this.toString()}.falseValue`,
       (typeOf(get(this, 'trueValue')) === 'undefined' && typeOf(get(this, 'falseValue')) === 'undefined') ||
       get(this, 'trueValue') !== get(this, 'falseValue'))
   },
 
-  // == Computed Properties =====================================================
-  @computed('trueValue', '_trueLabel')
-  _trueValue (trueValue, _trueLabel) {
-    return trueValue || _trueLabel
-  },
+  // == DOM Events ============================================================
 
-  @computed('falseValue', '_falseLabel')
-  _falseValue (falseValue, _falseLabel) {
-    return falseValue || _falseLabel
-  },
+  // == Lifecycle Hooks =======================================================
 
-  @computed('value')
-  _isToggled (value) {
-    return this._preferBoolean(value) === get(this, '_trueValue')
+  /* Ember.Component method */
+  init () {
+    this._super(...arguments)
+    this._setupAssertion()
   },
 
-  // == Actions ================================================================
+  // == Actions ===============================================================
+
   actions: {
     /* eslint-disable complexity */
+    // FIXME: eslint
     _onClick (event) {
       if (get(this, 'disabled')) return
 
-      if (!isSimpleClick(event)) {
+      if (!ViewUtils.isSimpleClick(event)) {
```

Comment:
So, when do we want to destructure vs not? I don't believe we have something like `rewire` that lets you swap out top-level functions for stubs within the module under test, so I think if there's any chance you'd want to stub the thing in a test, it's a better bet to not destructure it. That being said, I'm not sure if we ever want to stub something like `Ember.run.later` or `Ember.run.scheduleOnce` so I guess those are safe to destructure (not this comment, I realize, but I think `isSimpleClick` falls into the other category of something we *would* want to stub, since it might be harder in the test to generate click events that are or aren't simple.